src/pulsar_socket.erl

```diff
@@ -37,13 +37,27 @@
 %% exposed ONLY for mocking
 -export([internal_getopts/3]).
 
+%% Async socket process
+-export([ spawn_link_connect/3
+        , send_batch_message_packet_async/7
+        , ping_async/1
+        , pong_async/1
+        ]).
+%% `proc_lib' loop
+-export([init/2, loop/2, stop/1]).
```

Comment:
maybe just use gen_server.

Reply:
I took "inspiration" from `kpro_connection`, which is just a `proc_lib` loop..  :thinking:

Reply:
:checkered_flag: